### PR TITLE
Install Pod Restart Metric + Status

### PR DESCRIPTION
**Pull request by dgoodwin**
Builds on @ShivamChamoli 's PR and adds storing the count on status as well.

### CODE REVIEW

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 1]
```diff
@@ -902,6 +925,61 @@ func dnsZoneName(cdName string) string {
...
+func selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
```
Comment (csrwng) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go]: just a style suggestion (feel free to ignore) -- exit early here
if pod.Labels == nil {
     return retval
}

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 2]
```diff
@@ -902,6 +925,61 @@ func dnsZoneName(cdName string) string {
...
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
```
Comment (csrwng) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go]: same as above ... !ok should return early

File: pkg/controller/clusterdeployment/clusterdeployment_controller.go [hunk 3]
```diff
@@ -902,6 +925,61 @@ func dnsZoneName(cdName string) string {
...
+
+	if len(pods.Items) > 1 {
+		log.Warnf("found %d install pods for cluster", len(pods.Items))
+	}
+
+	// Calculate restarts across all containers in the pod:
+	podRestarts := 0
+	for _, pod := range pods.Items {
+		for _, cs := range pod.Status.ContainerStatuses {
+			podRestarts = podRestarts + int(cs.RestartCount)
```
Comment (csrwng) [inline pkg/controller/clusterdeployment/clusterdeployment_controller.go]: +=

**Comment by dgoodwin** [comment]
Updated. That integration test is failing a bit too much.

**Comment by csrwng** [comment]
@dgoodwin the e2e failure seems legit... the job is never created by the controller which tells me that the controller might be failing to start. Most likely due to a permissions issue. Did we add pods to the manager role?

**Comment by dgoodwin** [comment]
Nope I did not, thank you, should have noticed that was failing too much, will check if there's an rbac issue.

**Comment by csrwng** [comment]
@dgoodwin looks like you need to run `make generate` again

**Comment by csrwng** [comment]
nm, that passed
/lgtm
/retest

**Pull request merged**